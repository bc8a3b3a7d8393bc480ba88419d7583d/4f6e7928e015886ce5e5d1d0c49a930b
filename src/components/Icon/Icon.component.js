import { createIconSetFromFontello } from 'react-native-vector-icons';

import { config, name } from '../../assets/fontello';

const IconSet = createIconSetFromFontello(config, name);

export default IconSet;
