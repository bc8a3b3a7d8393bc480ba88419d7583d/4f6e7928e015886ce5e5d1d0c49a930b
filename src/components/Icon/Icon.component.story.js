import React from 'react';
import { FlatList, Text } from 'react-native';

// eslint-disable-next-line import/no-extraneous-dependencies
import { storiesOf } from '@storybook/react-native';

import { config } from '../../assets/fontello';

import Icon from './Icon.component';

const ListOfIcons = () => (
  <FlatList
    renderItem={({ item }) => (
      <Text style={{ color: 'black', fontSize: 30 }}>
        <Icon name={item.css} color="black" size={32} />
        {` ${item.css}`}
      </Text>
    )}
    data={config.glyphs}
    keyExtractor={item => item.css}
  />
);

storiesOf('Fontello icons', module).add('all icons', () => <ListOfIcons />);
