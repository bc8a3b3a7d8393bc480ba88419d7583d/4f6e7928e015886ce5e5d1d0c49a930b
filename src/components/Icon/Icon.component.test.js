import { testSnapshots } from '../../utils/test.util';

import Icon from './Icon.component';

describe('Fontello icon', () => {
  testSnapshots(Icon, [
    {
      description: 'basic render',
    },
  ]);
});
