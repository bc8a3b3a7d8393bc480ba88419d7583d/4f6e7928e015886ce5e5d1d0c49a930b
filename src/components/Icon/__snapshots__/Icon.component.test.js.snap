// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fontello icon snapshots basic render 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "4f6e7928e015886ce5e5d1d0c49a930b",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
/>
`;
